### PR TITLE
feat: add failover groups to sdk v2, and add data source for failover groups

### PR DESCRIPTION
**Pull request by sfc-gh-swinkler**
* add failover groups resource to v2 sdk
*  add a new data source for failover groups
 * add sweepers for cleanup purpose

### CODE REVIEW

**Comment by danu165** [comment]
Looking forward to using failover groups in terraform! Would you mind adding failover group grants as well? That would help enable managing failover groups completely in terraform.

**Comment by sfc-gh-swinkler** [comment]
> Looking forward to using failover groups in terraform! Would you mind adding failover group grants as well? That would help enable managing failover groups completely in terraform.
Sure i can do that.

**Comment by sfc-gh-swinkler** [comment]
@danu165 So i am looking at which grant resources would need to be added and I am not sure which grant you are referring to. From https://docs.snowflake.com/en/user-guide/account-replication-config#step-3-create-a-role-with-the-create-failover-group-privilege-in-the-target-account-optional we already have an account_grant resource that can grant "CREATE FAILOVER GROUP" to a role.
```
resource "snowflake_role" "test" {
  name = "myrole1"
}

resource "snowflake_account_grant" "test" {
  roles     = [snowflake_role.test.name]
  privilege = "CREATE FAILOVER GROUP"
}

```
However this is too much permissions if only using the data source. Which would require any of:

> FAILOVER
> 
> MONITOR
> 
> OWNERSHIP
> 
> REPLICATE

https://docs.snowflake.com/en/sql-reference/sql/show-failover-groups#usage-notes.

Is this what you mean?

**Comment by danu165** [comment]
Yes, I am referring to possibly creating a new resource named “snowflake_failover_group_grant” that could grant those 4 privileges that you mentioned.

File: pkg/sdk/sweepers.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		}
+		ctx := context.Background()
+		shares, err := client.Shares.Show(ctx, nil)
+		if err != nil {
+			return err
+		}
+		for _, share := range shares {
+			if share.Kind == ShareKindOutbound {
+				if prefix == "" || strings.HasPrefix(share.Name.Name(), prefix) {
+					if err := client.Shares.Drop(ctx, share.ID()); err != nil {
```
Comment (sfc-gh-ngaberel) [inline pkg/sdk/sweepers.go:73]: Would it make sense to merge those conditions into a single one to reduce indentation?

File: pkg/sdk/sweepers_test.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+)
+
+func TestSweepAll(t *testing.T) {
+	enableSweep := os.Getenv("SNOWFLAKE_ENABLE_SWEEP")
+	if enableSweep != "1" {
+		t.Skip("SNOWFLAKE_ENABLE_SWEEP not enabled, skipping sweep tests")
+	}
+
+	t.Run("all sweepers in secondary account", func(t *testing.T) {
+		client := testSecondaryClient(t)
```
Comment (sfc-gh-ngaberel) [inline pkg/sdk/sweepers_test.go:17]: Maybe create a few resources and check they're gone after?

File: pkg/sdk/failover_groups.go [hunk 3]
```diff
@@ -0,0 +1,565 @@
...
+	Add           *FailoverGroupAdd       `ddl:"keyword" db:"ADD"`
+	Move          *FailoverGroupMove      `ddl:"keyword" db:"MOVE"`
+	Remove        *FailoverGroupRemove    `ddl:"keyword" db:"REMOVE"`
+}
+
+func (opts *FailoverGroupAlterSourceOptions) validate() error {
+	if !validObjectidentifier(opts.name) {
+		return ErrInvalidObjectIdentifier
+	}
+	if !exactlyOneValueSet(opts.Set, opts.Add, opts.Move, opts.Remove) {
```
Comment (sfc-gh-ngaberel) [inline pkg/sdk/failover_groups.go:144]: NewName as well I suppose?

**Pull request merged**